Formula/core-lightning.rb

```diff
@@ -0,0 +1,41 @@
+class CoreLightning < Formula
+  desc "Lightning Network implementation focusing on spec compliance and performance"
+  homepage "https://github.com/ElementsProject/lightning"
+  url "https://github.com/ElementsProject/lightning.git",
```

Comment:
Do we need a Git clone? Can we use a tarball?

Reply:
The Git repo contains submodules. The build fails if they are not present.

Reply:
What are in the submodules?

Reply:
Dependencies
- gheap
- jsmn
- libbacktrace
- libsodium
- libwally-core
- lnprototest
- lowdown

Reply:
We should use Homebrew-provided dependencies where available (e.g. libsodium) instead of what's in the submodules.

Reply:
We are in the case of libsodium. The others are not in Homebrew.